experimental/PropertyDDS/packages/property-dds/src/propertyTreeExtFactories.ts

```diff
@@ -2,68 +2,107 @@
  * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
  * Licensed under the MIT License.
  */
+/* eslint-disable @typescript-eslint/no-unsafe-return */
 import { deflate, inflate } from "pako";
+import { compress, decompress } from "lz4js";
 import { bufferToString, stringToBuffer } from "@fluidframework/common-utils";
 import {
     IChannelAttributes,
     IFluidDataStoreRuntime,
     IChannelServices,
     IChannelFactory,
+    IChannel,
 } from "@fluidframework/datastore-definitions";
-import { IPropertyTreeMessage, ISharedPropertyTreeEncDec, ISnapshotSummary, SharedPropertyTreeOptions }
-from "./propertyTree";
-import { DeflatedPropertyTree } from "./propertyTreeExt";
-
-function encodeSummary(snapshotSummary: ISnapshotSummary) {
-    const summaryStr = JSON.stringify(snapshotSummary);
-    const unzipped = new TextEncoder().encode(summaryStr);
-    const serializedSummary: Buffer = deflate(unzipped);
-    return serializedSummary;
+import {
+    IPropertyTreeConfig, IPropertyTreeMessage, ISharedPropertyTreeEncDec,
+    ISnapshotSummary, SharedPropertyTree, SharedPropertyTreeOptions,
 }
+    from "./propertyTree";
+import { DeflatedPropertyTree, LZ4PropertyTree } from "./propertyTreeExt";
 
-function decodeSummary(serializedSummary): ISnapshotSummary {
-    const unzipped = inflate(serializedSummary);
-    const summaryStr = new TextDecoder().decode(unzipped);
-    const snapshotSummary: ISnapshotSummary = JSON.parse(summaryStr);
-    return snapshotSummary;
-}
+export abstract class CompressedPropertyTreeFactory implements IChannelFactory {
+    public abstract get attributes();
+    public abstract get type();
+    public abstract getEncodeFce();
+    public abstract getDecodeFce();
+    private createCompressionMethods(encodeFn, decodeFn): ISharedPropertyTreeEncDec {
+        return {
+            messageEncoder: {
+                encode: (change: IPropertyTreeMessage) => {
+                    const changeSetStr = JSON.stringify(change.changeSet);
+                    const unzipped = new TextEncoder().encode(changeSetStr);
+                    const zipped: Buffer = encodeFn(unzipped);
+                    const zippedStr = bufferToString(zipped, "base64");
+                    if (zippedStr.length < changeSetStr.length) {
+                        // eslint-disable-next-line @typescript-eslint/dot-notation
+                        change["isZipped"] = "1";
+                        change.changeSet = zippedStr;
+                    }
+                    return change;
+                },
+                decode: (transferChange: IPropertyTreeMessage) => {
+                    // eslint-disable-next-line @typescript-eslint/dot-notation
+                    if (transferChange["isZipped"]) {
+                        const zipped = new Uint8Array(stringToBuffer(transferChange.changeSet, "base64"));
```

Comment:
Just FYI - In the new SharedTree, we're using "changeset" rather than "changeSet".  (i.e., "changeset" is a single word, like "hashtable".)

However, 'changeSet' is okay if you prefer to be consistent with PropertyTree.

Reply:
This implementation is dedicated to Property DDS only and will not be used in any other context so I would suggest to stay consistent with Property DDS.